### PR TITLE
[plat-sam] sama5d2 improvements

### PR DESCRIPTION
**Pull request by clementleger**
This series embeds multiple improvements for the sama5d2 support. It mainly target at improving secure status for multiple devices as well as enabling reboot on watchdog reset.

### CODE REVIEW

**Comment by clementleger** [comment]
Update: Fix unwrapped commit

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>`

File: core/drivers/atmel_rstc.c [hunk 1]
```diff
@@ -43,6 +45,11 @@ static TEE_Result atmel_rstc_probe(const void *fdt, int node,
 {
 	size_t size = 0;
 
+	if (_fdt_get_status(fdt, node) != DT_STATUS_OK_SEC)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	matrix_configure_periph_secure(AT91C_ID_SYS);
```
Comment (ldts) [inline core/drivers/atmel_rstc.c:51]: is it not an issue if it fails? shouldn't we at least pop an error on the console?

Comment (clementleger) [reply core/drivers/atmel_rstc.c:51]: @ldts: If it fails, then the `atmel_rstc_available()` will return false. This is used by the code that can invoke reset support. So it is actually "supported", the same can happen if the rstc node is lacking.

**Comment by clementleger** [comment]
New version:
- Added  Acked-by: Etienne Carriere <etienne.carriere@linaro.org>

**Pull request merged**